rest/handler.go

```diff
@@ -209,17 +213,24 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		if dbContext, err = h.server.GetDatabase(dbname); err != nil {
 			base.Infof(base.KeyHTTP, "Error trying to get db %s: %v", base.MD(dbname), err)
 
-			if shouldCheckAdminAuth {
-				if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+			if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+				if shouldCheckAdminAuth {
 					authorized, err := h.checkAdminAuthenticationOnly()
 					if err != nil {
 						return err
 					}
 
-					if authorized {
-						return base.HTTPErrorf(http.StatusForbidden, "")
+					if !authorized || h.providedAuthCredentials() {
+						return ErrInvalidLogin
+					}
+				} else {
+					if h.privs == regularPrivs || h.privs == publicPrivs {
```

Comment:
Is this privs check required, if we're in the else for "shouldCheckAdminAuth"?

Reply:
This covers the case (which is maybe test-only) where h.server.config.API.AdminInterfaceAuthentication

Reply:
I see.  These conditions are organized in a way that's a bit hard to follow, but I guess the `else` on line 226 covers both "public port" and "admin port but admin auth is disabled", and the latter falls through to returning 'not found' on line 237.